### PR TITLE
Remove Void type

### PR DESCRIPTION
**Pull request by derobins**
* Remove Void (was: typedef to char *, now just void *)
* Remove NULL casts

### CODE REVIEW

File: mfhdf/src/nc_priv.h [hunk 1]
```diff
@@ -45,9 +45,6 @@
 #define NC_DOUBLE_SIZE      8
 #define NC_UNSPECIFIED_SIZE 0
 
-/* ptr argument type in internal functions */
-#define Void char
```
Comment (byrnHDF) [inline mfhdf/src/nc_priv.h:49]: WOW

**Review by byrnHDF — APPROVED** [review]

File: mfhdf/src/nc_priv.h (hunk 1 above)
Comment (derobins) [reply mfhdf/src/nc_priv.h:49]: HDF4 is always a journey down memory lane

Comment (derobins) [reply mfhdf/src/nc_priv.h:49]: At least this one is internal. The VOID macro is in the public API and HDF-EOS2 uses it.

**Review by bmribler — APPROVED** [review]

**Comment by derobins** [comment]
Closed in lieu of #686. Turns out Void is being used as a byte pointer, not a "generic object" pointer.

**Pull request closed without merging**